lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -46,6 +50,15 @@ def get_problem_scores(self, subsection):
         return problem_scores
 
     def get_url(self, subsection):
+        """
+        Returns the URL for the subsection while taking into account if the course team has
+        marked the subsection's visibility as hide after due.
+        """
+        hide_url_date = (subsection.self_paced and subsection.end) or subsection.due
```

Comment:
this isn't technically the hide_url_date if subsection.hide_after_due is False, not sure if it would be clearer or more confusing to do 
`hide_url_date = subsection.hide_after_due and ((subsection.self_paced and subsection.end) or subsection.due)`

Reply:
I prefer it as it is as the logic for setting that variable is only taking into account what matters for determining the date. I don't want to hide the other logic (hide_after_due and/or staff_access) into determining what date to use.